### DISCUSSION TITLE
how to resize, reposition, merge multiple images with fixed canvas width/height

### DISCUSSION THREAD

**Discussion by DOEHOONLEE** [ImageMagick/ImageMagick#4496] [Help]
Hi!

Let’s say there are three different pictures with different sizes and imagine there is a canvas that is 1000x1000 big. I would like to resize the picture A to `-resize 100x100!` and position it at `-geometry +200+150`. Likewise, I would like to resize and reposition the other two pictures as well.

I know I can resize/reposition a picture with a canvas size that I want like this:
`convert imageA.png -resize 100x100 -background transparent -extent 2000x2000-50-100 imageAinCanvas.png`

but I would like to do those in one command line with `clone`.

If anybody has a creative solution, please share! Thanks!!

**Comment by snibgo** [comment]
What version of IM, on what platform. I will assume IM v7 on Windows.

```
magick ^
  -size 1000x1000 xc:None ^
  ( imageA.png -resize "100x100^!" ) -geometry +200+150 -composite ^
  ( imageB.png -resize "100x100^!" ) -geometry +300+250 -composite ^
  out.png
```
Add as many input images as you want.

**Reply by DOEHOONLEE** [reply]
![Screen Shot 2021-11-25 at 9 52 23 AM](https://user-images.githubusercontent.com/20305442/143333148-777e7469-fd5c-48d9-8657-65bb9778b3b9.png)

I am on ImageMagick version 7.1 and on macOS. I ran the following commands with `^` switched to `\` assuming that `^` is for windows, but it's not working. Am I doing something wrong?

```
magick \ -size 1000x1000 xc:None \ (andie.png -resize 100x100!) -geometry +200+150 -compsite \ (rose.png -resize 300x170!) -geometry +100+50 -composite \ output.png
```

**Comment by snibgo** [comment]
You have:
```
 (andie.png -resize 100x100!) 
```
... but you must have a space before and after every "(" and every ")".
